grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcServiceImplErrorTest.java

```diff
@@ -0,0 +1,288 @@
+                    GrpcService
+                            .builder()
+                            .addService(
+                                    new ServerImplErrorAtMetadataService())
+                            .build();
+            sb.service(grpcService);
+            sb.decorator(LoggingService.newDecorator());
+        }
+    };
+
+    @RegisterExtension
+    static ServerExtension serverWithBlockingExecutor = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            final GrpcService grpcService =
+                    GrpcService.builder()
+                               .addService(
+                                       new ServerImplErrorAtMetadataService())
+                               .enableUnframedRequests(true)
+                               .supportedSerializationFormats(
+                                       GrpcSerializationFormats.values())
+                               .useBlockingTaskExecutor(true)
+                               .build();
+            sb.service(grpcService);
+            sb.decorator(LoggingService.newDecorator());
+        }
+    };
+    private static ManagedChannel channel;
+    private static ManagedChannel blockingChannel;
+
+    @BeforeAll
+    static void setUpChannel() {
+        channel = ManagedChannelBuilder
+                .forAddress("127.0.0.1", server.httpPort())
+                .usePlaintext()
+                .build();
+        blockingChannel = ManagedChannelBuilder
+                .forAddress("127.0.0.1", serverWithBlockingExecutor.httpPort())
+                .usePlaintext()
+                .build();
+    }
+
+    @AfterAll
+    static void tearDownChannel() {
+        channel.shutdownNow();
+        blockingChannel.shutdownNow();
+    }
+
+    // Normal case of #onError at Unary RPC.
+    @Test
+    void clientUnaryCall2() throws InterruptedException {
+        assertThatThrownBy(() -> {
+            TestServiceBlockingStub client = TestServiceGrpc.newBlockingStub(blockingChannel);
+            client.unaryCall2(REQUEST_MESSAGE);
+        }).satisfies(cause -> {
+            assertThat(Status.fromThrowable(cause).getCode()).isEqualTo(Status.ABORTED.getCode());
+            assertThat(Status.trailersFromThrowable(cause)).satisfies(metadata -> {
+                assertThat(metadata).isNotEqualTo(validMetadata);
+            });
+        });
+
+        RequestLog log = serverWithBlockingExecutor.requestContextCaptor().take().log().whenComplete().join();
+        assertThat(log.responseCause())
+                .isInstanceOf(StatusRuntimeException.class)
+                .satisfies(cause -> {
+                    assertThat(Status.fromThrowable(cause).getCode()).isEqualTo(Status.ABORTED.getCode());
+                    assertThat(Status.trailersFromThrowable(cause)).satisfies(metadata -> {
+                        assertThat(metadata).isEqualTo(validMetadata);
+                    });
```

Comment:
@ikhoon 

The `cause` of L131 and L142 returns different kind of metadata.
Was this a kind of what you afraid of at
https://github.com/line/armeria/pull/4686#discussion_r1124165001 ?

Please point out incorrect assertions in this unit test, as I've never used GRPC of Armeria... I'm sorry...